### PR TITLE
[AutoMM] Fix one lightning upgrade issue

### PR DESCRIPTION
**Pull request by zhiqiangdon**
*Issue #, if available:*

*Description of changes:*

1. Fix one lightning upgrade issue.
2. Bump up some lower caps.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by yinweisu** [comment]
| Previous CI Run | Current CI Run |
| :---: | :---: |

File: core/src/autogluon/core/_setup_utils.py [hunk 1]
```diff
@@ -26,9 +26,9 @@
     "networkx": ">=3.0,<4",  # Major version cap
     "tqdm": ">=4.38,<5",  # Major version cap
     "Pillow": ">=10.0.1,<11",  # Major version cap
-    "torch": ">=2.0,<2.2",  # "<{N+2}" upper cap, sync with common/src/autogluon/common/utils/try_import.py
-    "lightning": ">=2.0.0,<2.2",  # "<{N+2}" upper cap
-    "pytorch_lightning": ">=2.0.0,<2.2",  # "<{N+2}" upper cap, capping `lightning` does not cap `pytorch_lightning`!
+    "torch": ">=2.1,<2.2",  # "<{N+2}" upper cap, sync with common/src/autogluon/common/utils/try_import.py
+    "lightning": ">=2.1,<2.2",  # "<{N+2}" upper cap
+    "pytorch_lightning": ">=2.1,<2.2",  # "<{N+2}" upper cap, capping `lightning` does not cap `pytorch_lightning`!
```
Comment (prateekdesai04) [inline core/src/autogluon/core/_setup_utils.py:31]: nit {N+1}

File: core/src/autogluon/core/_setup_utils.py [hunk 2]
```diff
@@ -26,9 +26,9 @@
     "networkx": ">=3.0,<4",  # Major version cap
     "tqdm": ">=4.38,<5",  # Major version cap
     "Pillow": ">=10.0.1,<11",  # Major version cap
-    "torch": ">=2.0,<2.2",  # "<{N+2}" upper cap, sync with common/src/autogluon/common/utils/try_import.py
-    "lightning": ">=2.0.0,<2.2",  # "<{N+2}" upper cap
-    "pytorch_lightning": ">=2.0.0,<2.2",  # "<{N+2}" upper cap, capping `lightning` does not cap `pytorch_lightning`!
+    "torch": ">=2.1,<2.2",  # "<{N+2}" upper cap, sync with common/src/autogluon/common/utils/try_import.py
```
Comment (prateekdesai04) [inline core/src/autogluon/core/_setup_utils.py:29]: are we taking off support for Torch 2.0 completely ?

Comment (zhiqiangdon) [reply core/src/autogluon/core/_setup_utils.py:29]: Yes, it would install 2.1 anyway. Looks like no need to keep 2.0 here.

**Review by tonyhoo — APPROVED** [review]
Thanks. LGTM and please merge after tests passing

File: core/src/autogluon/core/_setup_utils.py (hunk 1 above)
Comment (zhiqiangdon) [reply core/src/autogluon/core/_setup_utils.py]: Updated

**Comment by yinweisu** [comment]
| Previous CI Run | Current CI Run |
| :---: | :---: |
| threadpoolctl==3.3.0 | threadpoolctl==3.4.0 |
| google-auth==2.28.2 | google-auth==2.29.0 |
| threadpoolctl==3.3.0 | threadpoolctl==3.4.0 |
| google-auth==2.28.2 | google-auth==2.29.0 |

**Pull request merged**